File: programmers/pg_lv2_영어끝말잇기.js
function solution(n, words) {
  let num = undefined;
  let order = undefined;
  let newArr = [];
  newArr.push(words[0]);

  for (i = 1; i < words.length; i++) {
    if (newArr[newArr.length - 1].slice(-1) != words[i].slice(0, 1) || newArr.includes(words[i])) {
      num = (i + 1) % n;
      order = parseInt(i / n) + 1;
      break;
    }
    newArr.push(words[i]);
  }

  if (num == undefined) return [0, 0];
  if (num == 0) return [n, order];
  return [num, order];
}

File: python/programmers/pg_lv2_영어끝말잇기.py
def solution(n, words):
    wordsSet = set()    
    
    for (idx, i) in enumerate(words):
        if idx == 0 or (i not in wordsSet and words[idx - 1][-1] == words[idx][0]): wordsSet.add(i)
        else:
            num = idx % n + 1
            turn = idx // n + 1
            return [num, turn]
    return [0,0]